src/main/java/org/opensearch/security/user/User.java

```diff
@@ -238,7 +238,7 @@ public final void copyRolesFrom(final User user) {
     public void writeTo(StreamOutput out) throws IOException {
         out.writeString(name);
         out.writeStringCollection(new ArrayList<String>(roles));
-        out.writeString(requestedTenant);
+        out.writeString(requestedTenant == null ? "" : requestedTenant);
```

Comment:
Could there be any side-effect of this change? Isn't empty string interpreted as global tenant?

Did you encounter any issue serializing a user while testing?

Reply:
When `requestedTenant` is null, call to `out.writeString(requestTenant)` results in a `NullPointerException`. Added this logic to handle the same. 

I'm not sure if an empty requestedTenant is interpreted as global tenant. If so, we can handle this in the `User(StreamInput in)` constructor.

Reply:
What steps did you take to get the NPE? Could this be an existing bug in the security plugin?

Reply:
```
User user = new User("user");
user.writeTo(out); // <- NPE
```

Reply:
I just checked through the codebase and it looks like null and empty string are being treated equally. Resolving this thread.